### ISSUE TITLE
Document or check that number of samples is large enough

### ISSUE THREAD

**Issue by Ardillen66** [braindecode/braindecode#432]
Hello,

I was experimenting with Deep4Net and came upon an undocumented issue. I tried using an input window of 64 channels by 375 samples and while this worked with ShallowFBCSPNet, I got errors from PyTorch about a mismatch in kernel size compared to the input (9x1 input for 10x1 kernel in this case). I tracked this down as a problem with the input_window_samples dimension that became too small in the 4th Conv layer of the model.

The solution to this problem was to either decrease the kernel size for the 4th Conv layer or to increase the number of input samples.

I suggest adding a note in the documentation to avoid users spending time trying to track down this specific issue. Maybe there could be a recommended minimal input size for each model. Ideally, there would also be a check that can give a sensible error to quickly resolve this issue. Another solution would be to introduce padding where necessary, but I am not familiar enough with the models to do this myself.

If I have the time, I will try to add this myself. But I cannot give any guarantees.

Thank you for the great work on this library!

**Comment by bruAristimunha** [comment]
Hi Arnau Dillen (@Ardillen66)!

I like your suggestions; I've been through something similar. Just a quick recap to see if I understood correctly:

1) Add some warning about sample sizes, try to check the minimum size, and establish some relationship for all 17 braindecode models and add this for the doc and the code.

2) improve error handling to try to warn the user that this kernel size issue  is usually related to the input size.

3) try to introduce padding mechanisms to the models (idk if this is a good idea),

Could you write some warning text? This would help the library and make it easier to implement the features. It is essential to have these multiple contributions to the braindecode to have a text that most people can understand.

**Comment by Ardillen66** [comment]
Hi,

Sounds good!

I will try to write some documentation with the warning when I have the time and make a pull request for the updated docs.

**Comment by bruAristimunha** [comment]
Thank you! let me know if you need any help.

**Comment by bruAristimunha** [comment]
Thank you! let me know if you need any help.

**Comment by bruAristimunha** [comment]
Hi @Ardillen66,

Do you need some assistence?

**Comment by Ardillen66** [comment]
Hi,

Sorry for the long silence. I haven't had the opportunity to take care of this yet. If this can wait I can take care of the documentation  this month (I hope) and we can start looking into padding where appropriate after.

**Comment by sliwy** [comment]
@bruAristimunha @PierreGtch as we will have `EEGModuleMixin` really soon I started to solve this issue ->
I am going to remove `get_output_shape` and `to_dense_prediction` as well as add `convert_to_regressor` method that is currently just copied over several examples. I am quite close to have a complete PR. I will submit tomorrow to braindecode master when #488 is merged.

**Comment by PierreGtch** [comment]
@sliwy yes! Let’s call and talk about it tomorrow, I also think we need something like that
